src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,61 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.CinemaHall;
+import mate.academy.util.HibernateUtil;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    private static final Logger log = LogManager.getLogger(CinemaHallDaoImpl.class);
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            log.error("Can`t save to DB cinemaHall: {}", cinemaHall, e);
+            throw new DataProcessingException("Can't insert cinemaHall " + cinemaHall, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<CinemaHall> get(Long id) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return Optional.ofNullable(session.get(CinemaHall.class, id));
+        } catch (Exception e) {
+            log.error("Can't get a cinemaHall by id: {}", id, e);
+            throw new DataProcessingException("Can't get a cinemaHall by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> getAllCinemaHallQuery
+                    = session.createQuery("from CinemaHall", CinemaHall.class);
+            return getAllCinemaHallQuery.getResultList();
+        }
```

Comment:
![Screenshot 2022-06-20 at 08 08 52](https://user-images.githubusercontent.com/60258373/174529275-e7f91f53-9562-4cc3-8a05-a21dc90df1d0.png)
Before you create PR with your homework, you need to go through the checklist under the task, and correct all the points described there. The mentor will not check the work until the checklist items are corrected